app/controllers/api/notes_controller.rb

```diff
@@ -3,28 +3,47 @@
 module Api
   class NotesController < ApplicationController
     rescue_from ActiveRecord::ActiveRecordError, with: :active_record_error
-
-    def index
-      session_notes
-      respond_to :json
-    end
-
     def create
-      args = params.require(:note).permit(:type, :title, :text, :items,
-                                          :id, :order)
+      args = note_params
       new_note = Note.create!(args)
       session_notes << new_note
       render json: new_note, status: :created
     end
 
+    def index
+      session_notes
+      respond_to :json
+    end
+
     def share
-      note = Note.find_by!(id: params[:note_id])
+      note = finded_note
       note.sessions << current_session
       redirect_to '/', status: :found
     end
 
+    def update
+      args = note_params.except(:id)
+      note = finded_note
+      note.update!(args)
+      render json: note, status: :ok
+    end
+
+    def destroy
+      finded_note.destroy
+      redirect_to '/', status: :ok
+    end
+
     private
 
+    def finded_note
+      Note.find_by!(id: params[:id])
+    end
```

Comment:
```suggestion
    def note
      @note ||= Note.find(params[:id])
    end
```

и кода сразу меньше будет, новые параметры в этот инстанс контроллера прийти не могут, можно спокойно мемоизировать